### ISSUE TITLE
FE: "Get updates from <system>" links should open in a new tab

### ISSUE THREAD

**Issue by JayDarnell** [department-of-veterans-affairs/va.gov-cms#11848]
## Describe the defect
On each VAMC System page, on the front end, there is a block with the heading Get updates from <system name> (Example: Get updates from Lovell Federal TRICARE health care) The links in this block all open in a new tab.
## To Reproduce

1. Visit https://www.va.gov/sheridan-health-care/
2. Scroll down to the bottom of the page
3. Click any of the links in the **Get updates from VA Sheridan health care** block
4. Notice how all the links open new tabs

## AC / Expected behavior
- [x] All external links in the "Get updates from <system name>"  that are not otherwise affected by the "external" exception list should Always have target="_blank", Always have `rel="noreferrer"
- [x] Accessibility review

## Screenshots
![Screenshot 2022-12-07 at 10 47 50 AM](https://user-images.githubusercontent.com/21045418/206225429-1639c5bd-dee4-4a87-9341-9447dd2c92a9.png)

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [x] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by briandeconinck** [comment]
This issue was identified during the Collaboration Cycle Staging Review for Lovell VAMC on 2/28/2023. Applicable Collaboration Cycle labels have been added to this ticket for tracking and dashboard purposes, rather than creating a duplicate ticket.

**Comment by jilladams** [comment]
From refinement: need to confirm if non-VA links _are_ required to open in same tab per Federal guidance / a11y style & content guidelines. @laflannery to confirm.

**Comment by jilladams** [comment]
https://design.va.gov/components/link/#behavior
> **Open in the same window except in certain instances.** Links should only open in a new tab if clicking the link will result in the user losing progress or data. Otherwise, links should open in the same window.

**Comment by laflannery** [comment]
Thank you @jilladams I knew I had found this somewhere!!

@kmariepat-cityfriends based on the VA guidance above the ticket description is accurate - all of the links in the "Get updates" section should be updated to open in the **same** tab.

**Comment by kmariepat-cityfriends** [comment]
Rolling into Sprint 87 to confirm the reason for the domain overide and ensure the approach outlined in this card is the preferred approach

**Comment by jilladams** [comment]
Current state of affairs: https://dsva.slack.com/archives/CBU0KDSB1/p1687446025748509?thread_ts=1687371270.904739&cid=CBU0KDSB1

**Design system:** Acked the tech issue re: subdomains and said (Alex Taker) that they'd ticket a docs update to reflect that.
**Platform:** Hasn't yet clarified the nature of the subdomains list and if it might go away. So: we need to keep it until we're informed otherwise. 

In our changes, then, Ryan's notes from Slack stand: 
* `target="_blank"` will be added to external links from that subdomain list, and to any links without va.gov or vets.gov domain, and those will open in a new tab
* This is in conflict with the DS notes re: same tab / new tab, but: DS is aware and owns fixing that in docs.

@ryguyk @laflannery I think we can go ahead to code review then, of [Max's PR](https://github.com/department-of-veterans-affairs/content-build/pull/1608).

**Comment by laflannery** [comment]
@jilladams I tried to review the PR you linked above but it wouldn't load - I don't know if it was because it was from a couple weeks ago and needs a new review instance?

**Comment by jilladams** [comment]
6/29/23 update: talked to Ryan, clarified the logic for what's in an "external" in bullets above. 

For all of the URLs in Max's PR: he's **overriding** the Platform logic that establishes what is an external link, that the Design System and Platform team have signed off on in our support thread. So: this becomes effectively a no-op, that we can't change until the Design System and Platform team get aligned on an alternate approach.

Under the hood: we should remove `target="_blank"` and in general: not try to specify that, in deference to the underlying rules and the design system.

@laflannery fyi. Closing this.

FYI @kmariepat-cityfriends

**Comment by jilladams** [comment]
We got updated guidance from Design system today: https://dsva.slack.com/archives/CBU0KDSB1/p1690487173146249?thread_ts=1687371270.904739&cid=CBU0KDSB1
> we would like to change this logic and enforce the rule of a link opening in the same window & tab and make teams have to explicitly open any link in a different tab if it fits the exception criteria that we already have
> "Open in the same window except in certain instances. Links should only open in a new tab if clicking the link will result in the user losing progress or data. Otherwise, links should open in the same window."
> The sub-domain logic should stand. That said, that list of sub-domains predates both Danielle and I so we're not sure what criteria were used to create it nor how up-to-date it is. But we'll manage that at some point. For now no change to that list is needed.
> External links should have `rel="noreferrer"` added per [USWDS guidance](https://designsystem.digital.gov/components/link/). But we should drop the `target="_blank"`.

So for this ticket: 
* removing `target="_blank"` is already done and correct
* we now need to add `rel="noreferrer"`
Reopening ticket to make it easy to find the history of what happened here, and let's track the newest change here.

**Comment by laflannery** [comment]
@jilladams I somehow get very lost here, could you clarify for me?

- I see that in the code you all removed the `target="_blank"` on all the links
- However on the front end on all the the external links (4 of the 5) this is being added back - which I believe is because of some JavaScript that is forcing this?
- All the discussion seems to be around clarifying VA subdomain behavior. Did we get a clear answer yet on what to do with external domains? Are we able to remove this forced behavior from external, non-VA domains?

**Comment by jilladams** [comment]
Yes, legitimately confusing. 

So: the flow on this is (as far as I understand it): 
* If a link doesn't use `va.gov` / `vets.gov`, OR if it's in the subdomain list, the existing logic will consider it "external." 
    * @humancompanion-usds has said: this treatment should stand, and DS will work on figuring out what to do about it long term, as well as document the nuances here in https://design.va.gov/components/link/#behavior.
* For now: teams should not be explicitly _forcing_ a new tab (`target="_blank"`), unless we're interrupting a flow. Teams _should_ be adding `rel="noreferrer"`.

With all of that said: you're right that we can set up our code correctly with no `target="_blank"` and using `rel="noreferrer"`, and that may then get overwritten by the big blast logic that is rewriting for anything that is not va.gov / vets.gov. I believe I understand that this bigger change needs more input from Matt Dingee / Content team before we modify it. So what we're doing on this ticket is setting up our files such that if the big blast logic changes, our templates will be behaving correctly.

**Comment by jilladams** [comment]
From refinement: 
1. We will put target blank back
2. noreferrer is already present but is now clear in the ACs
3. we will not tackle the text update right now. Laura to follow up on that with the content team.

**Comment by CherylEvans** [comment]
This ticket was recreated in va.gov-team # [64835](https://app.zenhub.com/workspaces/collaboration-cycle-reviews-61d89e9e5f4c120011a0ad37/issues/gh/department-of-veterans-affairs/va.gov-team/64835)

**Comment by laflannery** [comment]
Removing CC labels because ticket was recreated

**Comment by xiongjaneg** [comment]
@xiongjaneg To find YouTube ticket as partner tickets for next sprint

**Comment by eselkin** [comment]
**Question about the To Reproduce step:**
All links EXCEPT the `VA Sheridan health care operating status` link open in new tabs.
ALL links already have attribute `rel=noreferrer`
(not an AC All links EXCEPT `VA Sheridan health care operating status` have attribute `rel=noopener`)

Are we saying in this ticket that we want to remove the `VA Sheridan health care operating status` from opening in a new tab? It already opens in the same tab because it's in the same domain. 

Or, are we saying we want all links from this section that do open new tabs to replace each other in that same new tab?

Multiple comments are using "same" for different reasons, I think.

**Comment by laflannery** [comment]
@eselkin Ya this is confusing because of how complicated this got and how much back and forth. Let's see if I can make this clear:
1. `VA Sheridan health care operating status`
   1. Should open in the same tab
   2. Should NOT have `target="_blank"`
   3. Should NOT have `rel="noreferrer"`
2. All other links within the "Get Updates" section
   1. Should open in a NEW tab
   2. Should have `target="_blank"` attribute
   3. Should have `rel="noreferrer"` attribute

Regarding the `rel=noopener` attribute you asked about, I don't know about this because it is [not mentioned anywhere in the guidance](https://design.va.gov/content-style-guide/links#linking-to-external-sites), I have never heard anyone talk about this. However, I did a very brief search and I found that we probably don't need this anywhere because implicitly I think it's the same as `target="_blank" (I think?). I got this from [the blue box on the MDN article here](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/noopener). If I'm reading that right, I think we can remove `rel=noopener` from everywhere.

**Comment by eselkin** [comment]
So basically, the only change is removing `rel=noreferrer` from the VA link in that section. 

And, right, noopener is antiquated because of how _blank now works on most browsers.

**Comment by jilladams** [comment]
Working on sprint report so went ahead and verified: all external domains open in a new tab, Operating status points to VA.gov and opens in same tab. Closing!
